packages/components/src/image-block/view/component.ts

```diff
@@ -142,7 +157,7 @@ export const imageComponent: Component<ImageComponentProps> = ({
         />
         <div class=${clsx('placeholder', hidePlaceholder && 'hidden')}>
           <input disabled=${readonly} class="hidden" id=${uuid} type="file" accept="image/*" onchange=${onUpload} />
-          <label onpointerdown=${onClickUploader} class="uploader" for=${uuid}>
+          <label onpointerdown=${onClickUploader} onclick=${handleUpload} class="uploader" for=${uuid}>
```

Comment:
I don't think we should add a new event handler here to handle both `onclick` and `onpointerdown`, a possible solution should be still use `onUpload` without introducting another config. We can pass `event` to the `onUpload` config to let users open their own file picker.